### DISCUSSION TITLE
Dating app "platform" that makes it easy to pivot/find niches

### DISCUSSION THREAD

**Discussion by ethanblake4** [PullDating/Org#38] [Ideas & Features]
I think it'd make sense to design the code as more of a "platform" that can quickly be branched off to create new apps in different niches. For the flutter side, we'd have a core Dart package with the API and some model code, as well as a core Flutter packages with some reusable UI components. On the server we could probably used a shared database server with a structure that's document-based or with relational key-value metadata instead of defined table structures, that has different databases (on the same server). This would make it easy to branch off and quickly create multiple apps with low overhead to find multiple niches and create less of a pressure to find a single great idea that will definitely be successful.

**Comment by ajbott** [comment]
I think this would be great as a plugin idea
-Library of different user created plugins for the app

Examples
-Filtering algorithms
-Different ways to match
-Different profile elements (viewing users would have to add plugin to view it on their profile)
